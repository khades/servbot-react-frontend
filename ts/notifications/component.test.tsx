import * as React from "react";
import { create } from "react-test-renderer";
import Notifications, { INotificationsProps } from "./component";

describe("Notifications", () => {
    it("should render properly  ", () => {
        const props: INotificationsProps = {
            hideNotification: (id) => id,

            notifications: [{
                body: "AAAAAA",
                date: new Date(),
                id: "222",
            }, {
                body: "BBBB",
                date: new Date(),
                id: "333",
            }, {
                body: "CCCCCCC",
                date: new Date(),
                id: "444",
            }],
        };
        const notifications = create(
            <Notifications {...props} />,
        );
        expect(notifications.toJSON()).toMatchSnapshot();

        const newProps: INotificationsProps = {
            hideNotification: (id) => id,

            notifications: [{
                body: "AAAAAA",
                date: new Date(),
                id: "222",
            }, {
                body: "BBBB",
                date: new Date(),
                id: "333",
            }],
        };
        notifications.update(
            <Notifications {...newProps} />);
        expect(notifications.toJSON()).toMatchSnapshot();

    });

    it("should properly hide notificaiton", () => {
        let hiddenNotification = "";
        const notificationToHide = "333";
        const props: INotificationsProps = {
            hideNotification: (id) => hiddenNotification = id,

            notifications: [{
                body: "AAAAAA",
                date: new Date(),
                id: "222",
            }, {
                body: "BBBB",
                date: new Date(),
                id: "333",
            }, {
                body: "CCCCCCC",
                date: new Date(),
                id: "444",
            }],
        };
        const notifications = create(
            <Notifications {...props} />,
        );
        const notification = notifications.root.findAll((el) => {
            return el.props["data-id"] === notificationToHide;
        });
        notification[0].props.onClick({ currentTarget: { dataset: { id: notificationToHide } } });
        expect(hiddenNotification).toEqual(notificationToHide);
    });
});
