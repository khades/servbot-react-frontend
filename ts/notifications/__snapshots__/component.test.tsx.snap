// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notifications should render properly   1`] = `
<div
  className="notifications"
>
  <div
    className="notifications__item"
    data-id="222"
    onClick={[Function]}
  >
    AAAAAA
  </div>
  <div
    className="notifications__item"
    data-id="333"
    onClick={[Function]}
  >
    BBBB
  </div>
  <div
    className="notifications__item"
    data-id="444"
    onClick={[Function]}
  >
    CCCCCCC
  </div>
</div>
`;

exports[`Notifications should render properly   2`] = `
<div
  className="notifications"
>
  <div
    className="notifications__item"
    data-id="222"
    onClick={[Function]}
  >
    AAAAAA
  </div>
  <div
    className="notifications__item"
    data-id="333"
    onClick={[Function]}
  >
    BBBB
  </div>
  <div
    className="notifications__item notifications__item--hidden"
    data-id="444"
    onClick={[Function]}
  >
    CCCCCCC
  </div>
</div>
`;
